Iced applications fail to compile under wasm target on master

On current master, Iced applications fail to compile under the wasm target, because `Subscription` is not implemented in `iced_web`:

```
error[E0432]: unresolved import `crate::Subscription`
 --> C:\Dev\iced\src\application.rs:1:41
  |
1 | use crate::{Command, Element, Settings, Subscription};
  |                                         ^^^^^^^^^^^^ no `Subscription` in the root

error[E0432]: unresolved import `crate::Subscription`
 --> C:\Dev\iced\src\sandbox.rs:1:54
  |
1 | use crate::{Application, Command, Element, Settings, Subscription};
  |                                                      ^^^^^^^^^^^^ no `Subscription` in the root
```

Maybe there could be a wasm build in the GitHub actions to highlight these issues?

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -1,7 +1,7 @@
 name: Test
 on: [push, pull_request]
 jobs:
-  all:
+  native:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
@@ -16,3 +16,14 @@ jobs:
       run: |
         cargo test --verbose --all
         cargo test --verbose --all --all-features
+
+  web:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: hecrj/setup-rust-action@v1
+      with:
+        rust-version: stable
+        targets: wasm32-unknown-unknown
+    - uses: actions/checkout@master
+    - name: Run checks
+      run: cargo check --package iced --target wasm32-unknown-unknown

--- a/src/application.rs
+++ b/src/application.rs
@@ -146,12 +146,12 @@ pub trait Application: Sized {
     /// It should probably be that last thing you call in your `main` function.
     ///
     /// [`Application`]: trait.Application.html
-    fn run(settings: Settings)
+    fn run(_settings: Settings)
     where
         Self: 'static,
     {
         #[cfg(not(target_arch = "wasm32"))]
-        <Instance<Self> as iced_winit::Application>::run(settings.into());
+        <Instance<Self> as iced_winit::Application>::run(_settings.into());
 
         #[cfg(target_arch = "wasm32")]
         <Instance<Self> as iced_web::Application>::run();

--- a/web/Cargo.toml
+++ b/web/Cargo.toml
@@ -15,7 +15,7 @@ categories = ["web-programming"]
 maintenance = { status = "actively-developed" }
 
 [dependencies]
-iced_core = { version = "0.1.0", path = "../core", features = ["command"] }
+iced_core = { version = "0.1.0", path = "../core", features = ["command", "subscription"] }
 dodrio = "0.1.0"
 wasm-bindgen = "0.2.51"
 wasm-bindgen-futures = "0.4"

--- a/web/src/hasher.rs
+++ b/web/src/hasher.rs
@@ -1,6 +1,6 @@
 use std::collections::hash_map::DefaultHasher;
 
-/// The hasher used to compare layouts.
+/// The hasher used to compare subscriptions.
 #[derive(Debug)]
 pub struct Hasher(DefaultHasher);
 

--- a/web/src/hasher.rs
+++ b/web/src/hasher.rs
@@ -1,0 +1,21 @@
+use std::collections::hash_map::DefaultHasher;
+
+/// The hasher used to compare layouts.
+#[derive(Debug)]
+pub struct Hasher(DefaultHasher);
+
+impl Default for Hasher {
+    fn default() -> Self {
+        Hasher(DefaultHasher::default())
+    }
+}
+
+impl core::hash::Hasher for Hasher {
+    fn write(&mut self, bytes: &[u8]) {
+        self.0.write(bytes)
+    }
+
+    fn finish(&self) -> u64 {
+        self.0.finish()
+    }
+}

--- a/web/src/lib.rs
+++ b/web/src/lib.rs
@@ -61,18 +61,22 @@ use std::{cell::RefCell, rc::Rc};
 
 mod bus;
 mod element;
+mod hasher;
 
 pub mod style;
+pub mod subscription;
 pub mod widget;
 
 pub use bus::Bus;
 pub use dodrio;
 pub use element::Element;
+pub use hasher::Hasher;
 pub use iced_core::{
     Align, Background, Color, Command, Font, HorizontalAlignment, Length,
     VerticalAlignment,
 };
 pub use style::Style;
+pub use subscription::Subscription;
 pub use widget::*;
 
 /// An interactive web application.

--- a/web/src/subscription.rs
+++ b/web/src/subscription.rs
@@ -1,0 +1,19 @@
+//! Listen to external events in your application.
+use crate::Hasher;
+
+/// A request to listen to external events.
+///
+/// Besides performing async actions on demand with [`Command`], most
+/// applications also need to listen to external events passively.
+///
+/// A [`Subscription`] is normally provided to some runtime, like a [`Command`],
+/// and it will generate events as long as the user keeps requesting it.
+///
+/// For instance, you can use a [`Subscription`] to listen to a WebSocket
+/// connection, keyboard presses, mouse events, time ticks, etc.
+///
+/// [`Command`]: ../struct.Command.html
+/// [`Subscription`]: struct.Subscription.html
+pub type Subscription<T> = iced_core::Subscription<Hasher, (), T>;
+
+pub use iced_core::subscription::Recipe;

